test/helpers/envtest.go

```diff
@@ -191,29 +204,15 @@ func NewTestEnvironment() *TestEnvironment {
 		klog.Fatalf("failed to set up webhook with manager for HCloudRemediationTemplate: %s", err)
 	}
 	// Create a fake HCloudClientFactory
-	hcloudClientFactory := fakeclient.NewHCloudClientFactory()
-
-	rescueSSHClient := &sshmock.Client{}
```

Comment:
you remove all of this from the init function. Why? If we do that, we should specify a reason why we don't initialize - probably in the function description

Reply:
I updated the docstring:

```go
// NewTestEnvironment creates a new environment spinning up a local api-server. Factories for
// clients (like HCloudClientFactory) are not created. This gets done in ResetAndCreateNamespace(),
// which should be use at the beginning of each test.
func NewTestEnvironment() *TestEnvironment {
```

ok like this?

Reply:
you have a Reset function which initializes a new namespace with the reconciler. You call this Reset function in another function ResetAndCreateNamespace. In theory the function should be ResetAndInitNamespace and not Reset.

Reply:
I renamed Reset() to ResetAndInitNamespace().